cryptoapis/src/endpoint/stats.ts

```diff
@@ -1,24 +1,32 @@
 import { Requester, Validator } from '@chainlink/external-adapter'
 import { AdapterRequest, Config } from '@chainlink/types'
+import { DEFAULT_ENDPOINT } from '../config'
 
-export const Name = 'difficulty'
+export const Name = 'stats'
 
-const difficultyParams = {
+const statsParams = {
   blockchain: ['blockchain', 'coin'],
+  endpoint: false,
```

Comment:
We should not reuse the name "endpoint" in our internal API. It just does not fit, just as it does not fit in the external API, but we are currently stuck with it there because of backward compatibility.

Internally, we should structure this differently.

So let's take this adapter and endpoint as an example, the "stats" endpoint.
The stats endpoint impl should have no knowledge of a concept like "height", this is a concept of our external API, it has nothing to do with "stats" endpoint. We should map a CL "height" request to a Cryptoapis "stats" request, but the impl for Cryptoapis "stats" request should have no knowledge of the outside CL "height" request.

The input to "stats" endpoint should look something like this:

```javascript
const inputParams = {
  blockchain: true,
  network: false,
  path: false  
}
```

It makes a "stats" requests, and returns the data point at "path". If the "path" is empty, it returns the whole response body.

The component that is calling this should map from the CL API request to the "stats" API request. So your adapter in the top lvl execute function gets the input:

```json
{
  "blockchain": "btc",
  "endpoint": height
}
```

We should map that to the underlying API endpoint fn call like this:

```json
{
  "blockchain": "bitcoin",
  "network": "mainnet",
  "path": "payload.blocks"
}
```

@ebarakos @RodrigoAD @justinkaseman @emmick4 please acknowledge & discuss!

As this is a high priority for today, you can ignore this for this specific PR, but keep it in mind on the next one. I am also going to open an issue for it, to improve our older implementations.